lib/mongo/client.rb

```diff
@@ -325,16 +325,25 @@ def initialize(addresses_or_uri, options = nil)
       if sdam_proc
         sdam_proc.call(self)
       end
+      @server_selection_semaphore = Semaphore.new
+      @cluster = Cluster.new(addresses, @monitoring, cluster_options)
+      # Unset monitoring, it will be taken out of cluster from now on
+      remove_instance_variable('@monitoring')
+      yield(self) if block_given?
+    end
+
+    # @api private
+    def cluster_options
       # We share clusters when a new client with different CRUD_OPTIONS
       # is requested; therefore, cluster should not be getting any of these
       # options upon instantiation
-      cluster_options = @options.reject do |key, value|
+      options.reject do |key, value|
```

Comment:
Would caching this value in an instance variable cause incorrect behavior? I don't think the options should change

Reply:
We do it all over the place in the driver already. What's different about here?

Reply:
It may be a premature optimization elsewhere as well.

Reply:
I don't feel strongly enough about this to block merging, so it's fine to leave it as-is